pkg/autoscaler/event.go

```diff
@@ -63,7 +63,11 @@ func newEventWatcher(clientset kubernetes.Interface) *eventWatcher {
 		},
 	})
 
-	return &w
+	if err != nil {
+		return nil, err
```

Comment:
Always wrap errors please
```suggestion
		return nil, fmt.Errorf("could not add event handler: %w", err)
```